### ISSUE TITLE
Being spammed Atlassian login

### ISSUE THREAD

**Issue by Frotty** [desktop/desktop#15162]
### The problem

After the latest update I am being spammed top login to atlassian even though I don#t actively use any atlassian repositories.
How can I turn off this stupid dialog popping up every 10 seconds??

### Release version

Version 3.0.6 (x64)

### Operating system

Windows 10

### Screenshots

![image](https://user-images.githubusercontent.com/1486037/186432339-278aee2a-2ae8-46f3-9876-91fd675e0d6d.png)

**Comment by sergiou87** [comment]
I'm sorry but I don't even know where is that popup coming from 😐 

Do you have any Atlassian repos in your repository list? The app might be fetching them, but nothing has changed in that regard 🤔  Maybe Atlassian changed something about how to authenticate to work with their repos from `git`, or whatever token/password you used expired right now and you need to log in again.

In any case, could you share app logs in case there is anything relevant there? 🤞

**Comment by Frotty** [comment]
Yes there is an error and I have some bitbucket repos in my big list of repos in the app which I didn't touch in forever, dunno why it would try to login to those when I'm not even using them? (before this version it did not do that..)

```
stderr:
remote: The requested repository either does not exist or you do not have access. If you believe this repository exists and you have access, make sure you're authenticated.
fatal: unable to access 'https://bitbucket.org/xxx/xxx.git/': The requested URL returned error: 403

(The error was parsed as 3: Authentication failed. Some common reasons include:

- You are not logged in to your account: see File > Options.
- You may need to log out and log back in to refresh your token.
- You do not have permission to access this repository.
- The repository is archived on GitHub. Check the repository settings to confirm you are still permitted to push commits.
- If you use SSH authentication, check that your key is added to the ssh-agent and associated with your account.
- If you use SSH authentication, ensure the host key verification passes for your repository hosting service.
- If you used username / password authentication, you might need to use a Personal Access Token instead of your account password. Check the documentation of your repository hosting service.)
2022-08-24T09:26:28.591Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 1.431s)
2022-08-24T09:26:28.591Z - error: [ui] `git -c credential.helper= fetch --progress --prune origin` exited with an unexpected code: 128.

```

**Comment by sergiou87** [comment]
> before this version it did not do that..

Nothing has change in this version that could alter that behavior, so my bet is the credentials you had stored to access those repos just expired, and that's why it's showing you that popup 😕 

> I have some bitbucket repos in my big list of repos in the app which I didn't touch in forever, dunno why it would try to login to those when I'm not even using them

You might have seen that the list of repositories sometimes show a small indicator of whether or not there are new commits in the current branch of each repo. That's because GitHub Desktop performs a git fetch in background for repos: https://github.com/desktop/desktop/blob/a1af96c131a5dd3532c13c2176141cd50ca5358d/app/src/lib/stores/app-store.ts#L3252

My advice is that you either login with that Atlassian popup, or just remove those repos from your list since you don't use them (it will also make your PC use less resources by avoiding unnecessary `git fetch`).

In summary, I don't think this is a bug, just a coincidence with the new release 😕

**Comment by SamLefebvre** [comment]
I got the same popup while I tried to push my last commit and I got an Atlassian popup while it was pushing my update.
I did not connect to that repo for the last 2 years,  it was under a vpn so it did not connect to it before automatically.

Here the error I got today.
```
2022-08-24T19:20:26.316Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 11.242s)
2022-08-24T19:20:26.326Z - error: [ui] `git -c credential.helper= fetch --progress --prune origin` exited with an unexpected code: 128.
stderr:
fatal: unable to access 'http://gitlab-ce.xxx.ca/xxx.git/': Could not resolve host: gitlab-ce.xxx.ca
```

Here is my log from last week, I got the same error. I did not had any popup at that time.

```
2022-08-13T21:02:18.233Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 11.274s)
2022-08-13T21:02:18.273Z - error: [ui] `git -c credential.helper= fetch --progress --prune origin` exited with an unexpected code: 128.
stderr:
fatal: unable to access 'http://gitlab-ce.xxx.ca/xxx.git/': Could not resolve host: gitlab-ce.xxx.ca
```

I got the Atlassian popup only one time, it did not appear for later push.

**Comment by Frotty** [comment]
> In summary, I don't think this is a bug, just a coincidence with the new release 😕

I think you are wrong, as downgrading makes the this behavior disappear 👍

**Comment by sergiou87** [comment]
@Frotty hah! Interesting. Which version did you downgrade to? Just 3.0.5?

If so, looking at the changes since 3.0.5, https://github.com/desktop/desktop/pull/14985 looks like the most probable culprit 🤔

**Comment by sergiou87** [comment]
@Frotty as mentioned in https://github.com/desktop/desktop/issues/15163#issuecomment-1227362552

…do you know which specific repository is showing that dialog to you? **It might not be the repository you're in, but some other from your repository list**.

Once you find out which one is showing the popup, it'd be nice if you could open it from a terminal and run `git symbolic-ref -q refs/remotes/origin/HEAD`. I think that's the command causing the popup (added in #14985 ). If you manage, to show the popup from the terminal running that command, try again running this other one: `git -c credential.helper= symbolic-ref -q refs/remotes/origin/HEAD`

If my theory is correct, that `-c credential.helper=` should fix the problem.

**Comment by sergiou87** [comment]
@Frotty you can also try out ~~this test build~~ which adds `-c credential.helper=` to the aforementioned `git` command introduced in #14985 . Please let me know if that fixes the problem 🤞 

**EDIT:** removed the link, this is the right test build: https://centralreleases.blob.core.windows.net/github-desktop/releases/3.0.7-test2-1da95508/GitHubDesktopSetup-x64.exe

**Comment by Manttipaa** [comment]
I have the same problem. I tried to test the build you linked (3.0.7-test1) and same popup still appeared multiple times.

(I deleted old repos after this test. Seems like the popup stopped appearing after.)

**Comment by sergiou87** [comment]
I'm sorry @Manttipaa and @Frotty , I made the change in the wrong git command (it's not `symbolic-ref` the one doing the network call, but `remote set-head`).

This one should contain the proper fix: https://centralreleases.blob.core.windows.net/github-desktop/releases/3.0.7-test2-1da95508/GitHubDesktopSetup-x64.exe 🤞

**Comment by justingolden21** [comment]
Is this being looked into? I also get this annoying popup but it's more like every hour than every few seconds (thankfully...) I have some repos on bitbucket and we require VPN to access them. Sometimes I am on VPN and sometimes I'm not. But the kicker is that I haven't opened any of those repos on Desktop or anywhere in **weeks** and yet I somehow keep getting these popups when I haven't even changed repos from my current (gitlab) repo in **weeks**. Must be the recent update.

**Comment by sergiou87** [comment]
@justingolden21 please try this build which should fix the issue and let me know: https://centralreleases.blob.core.windows.net/github-desktop/releases/3.0.7-test2-1da95508/GitHubDesktopSetup-x64.exe

Thank you!

**Comment by sergiou87** [comment]
I'm gonna close this in favor of #15163 , which has more info. Thanks again for your report! 🙇‍♂️

**Comment by sergiou87** [comment]
Hi all! We just released 3.0.7 with the fix a few hours ago.

> **Warning**
> If you have installed either the latest beta or one of the test builds I shared (3.0.7-test1 and 3.0.7-test2), **DO NOT** install 3.0.7 unless you don't mind losing your repositories on GitHub Desktop and re-adding them there. They **won't** be removed from your local disk, don't worry about that, they're just removed from the **repository list** inside of GitHub Desktop. This is because those beta and test builds use [Electron](https://electronjs.org) v19, while the latest 3.0.7 has Electron v17 and won't be able to open the repository database once it's been migrated to Electron v19.
>
> We will release a new production build soon with Electron v19 that you can update to. I will let you know again when that happens.

If ☝️ this ☝️ warning doesn't affect you, or you don't mind re-adding all your repos to GitHub Desktop, you can update immediately from the About dialog. If it doesn't find a release (**it's not enabled to _all_ users yet**), try hold <kbd>Alt</kbd> on your keyboard and then click on the `Ensure Last Version` button to force the upgrade:

https://user-images.githubusercontent.com/1083228/187416390-01ee032d-8dca-4635-ab0e-a3c4c87e4c5d.mov

You can also download it from https://desktop.github.com

Thank you for your patience!! 🙇‍♂️
